tests/Baseline/spicy.optimization.default-parser-functions/noopt.hlt

```diff
@@ -61,6 +61,12 @@ public type P2 = struct {
     method tuple<view<stream>, int<64>, iterator<stream>> __parse_foo_P2_stage2(inout value_ref<stream> __data, copy view<stream> __cur, copy bool __trim, copy int<64> __lah, copy iterator<stream> __lahe);
 } &on-heap;
 
+global auto __feat%foo__P1%supports_filters = True;
```

Comment:
The `%` inside the IDs feel a bit funky. :) Just use underscores instead?

Reply:
I went with `%` instead of `_` since we already encode the `::` in type names as `__`; using the same token to separate different components makes extraction (which we still do in too many places) slightly tricky. I'd prefer to leave this as is until we possibly replace these global constants with dedicated `Feature` AST types later like you suggested.